code/modules/client/preferences/entries/character/names.dm

```diff
@@ -127,34 +150,37 @@
 /datum/preference/name/religion
 	db_key = "religion_name"
 
-	allow_numbers = TRUE
+	group = "z_religion" // should be after silicon name group
```

Comment:
if you want a specific order, use number defines rather than alphabetical sorting. (e.g. order = PREFERENCE_NAME_ORDER_RELIGION)